comp/logs/agent/config/endpoints.go

```diff
@@ -41,8 +41,9 @@ type Endpoint struct {
 
 	Host                    string `mapstructure:"host" json:"host"`
 	Port                    int
-	UseCompression          bool `mapstructure:"use_compression" json:"use_compression"`
-	CompressionLevel        int  `mapstructure:"compression_level" json:"compression_level"`
+	UseCompression          bool   `mapstructure:"use_compression" json:"use_compression"`
+	CompressionKind         string `mapstructure:"compression_kind" json:"compression_kind"`
+	CompressionLevel        int    `mapstructure:"compression_level" json:"compression_level"`
```

Comment:
Should we also have `zstd_compression_level` here? (and in the endpoint constructor below)

Reply:
`CompressionLevel` gets set to either `compression_level` or `zstd_compression_level` earlier on in the process since there's not a need to have two separate fields at this stage.

Reply:
Makes sense. I forget that `additional_endpoints` inherit the compression strategy from the main endpoint. 


However this opens up a followup question, is there a use case where an additional endpoint would use a different compression strategy than the main endpoint? (ex if an on-prem log collector/proxy doesn't support zstd, and we enable zstd by default for the main endpoint). 

We can probably track this as a followup.

Reply:
It's a good question. I think we have been looking towards only supporting one compression strategy throughout the agent (both metrics and logs). It just makes things simpler. So if there is a case where we should allow multiple at the same time, we should find out about these now.